Go: More Views examples (#3285)
open-telemetry/opentelemetry.io: content/en/docs/instrumentation/go/manual.md

)
```

> The SDK filters metrics and attributes before exporting metrics. This allows
> views to be used to e.g. reduce memory usage of high cardinality metrics or drop
> attributes that can contain sensitive data.

Reviewer: ```suggestion
The SDK filters metrics and attributes before exporting metrics. For example, you
can use views to reduce memory usage of high cardinality metrics or drop
attributes that might contain sensitive data.
```

> For example, here's how you create a view that drops the `latency` instrument
> from the `http` instrumentation library:

Reviewer: ```suggestion
Here's how you create a view that drops the `latency` instrument
from the `http` instrumentation library:
```

```go
view := metric.NewView(
  metric.Instrument{
    Name:  "latency",
    Scope: instrumentation.Scope{Name: "http"},
  },
  metric.Stream{Aggregation: metric.AggregationDrop{}},
)

meterProvider := metric.NewMeterProvider(
	metric.WithView(view),
)
```

> For example, here's how you create a view removes the `http.request.method`

Reviewer 2: ```suggestion
For example, here's how you create a view that removes the `http.request.method`
```

> For example, here's how you create a view removes the `http.request.method`

Reviewer 2: The `For example` here is a bit repetitive with the one above. Suggestion:

```suggestion
As a second example, here's how you create a view removes the `http.request.method`
```

> For example, here's how you create a view that removes the `http.request.method`
> attribute recorded by the `latency` instrument from the `http` instrumentation
> library:

Reviewer: ```suggestion
Here's how you create a view that removes the `http.request.method`
attribute recorded by the `latency` instrument from the `http` instrumentation
library:
```

```go
view := metric.NewView(
  metric.Instrument{
    Name:  "latency",
    Scope: instrumentation.Scope{Name: "http"},
  },
  metric.Stream{AttributeFilter: attribute.NewDenyKeysFilter("http.request.method")},
)

meterProvider := metric.NewMeterProvider(
	metric.WithView(view),
)
```

> The `Name` field of criteria supports wildcard pattern matching. The wildcard
> `*` is recognized as matching zero or more characters, and `?` is recognized as
> matching exactly one character. For example, a pattern of `*` will match all
> instrument names.
> 
> For example, here's how you create a view that sets unit to milliseconds for any
> instrument with a name suffix of `.ms`:

Reviewer: ```suggestion
The `Name` field supports wildcard pattern matching. The `*` wildcard
is recognized as matching zero or more characters, and `?` is recognized as
matching exactly one character. For example, a pattern of `*` matches all
instrument names.

The following example show how you create a view that sets unit to
milliseconds for any instrument with a name suffix of `.ms`:
```
